apps/model-diagnostics/model_diagnostics.cpp

```diff
@@ -0,0 +1,52 @@
+/*************************************************
+USAGE:
+./model_diagnostics -m <onnx file location>
+**************************************************/
+#include <opencv2/dnn.hpp>
+#include "../../samples/dnn/common.hpp"
+
+#include <fstream>
+#include <iostream>
+
+
+using namespace cv;
+using namespace dnn;
+
+
+static void diagnosticsErrorCallback(const Exception& exc)
+{
+    fflush(stdout); fflush(stderr);
+}
+
+std::string diagnosticKeys =
+        "{ model m     | | Path to the model .onnx file. }"
+        "{ config c    | | Path to the model configuration file. }"
+        "{ framework f | | [Optional] Name of the model framework. }";
+
+
+
+int main( int argc, const char** argv )
+{
+    CommandLineParser argParser(argc, argv, diagnosticKeys);
+    argParser.about("Use this tool to run the diagnostics of provided ONNX model"
+                    "to obtain the information about its support (supported layers).");
+
+    if (argc == 1)
+    {
+        argParser.printMessage();
+        return 0;
+    }
+
+    String model = findFile(argParser.get<String>("model"));
+    String config = findFile(argParser.get<String>("config"));
+    String frameworkId = argParser.get<String>("framework");
+
+    CV_Assert(!model.empty());
+
+    setDiagnosticsRun(true);
```

Comment:
it's not good to alter DNN behaviour by setting a global variable. It would be better to have an overloaded variant of readNet() that will pass the corresponding flags to the importers w/o affecting global state.

Reply:
I prefer adding global variable instead of exploding of public API.
API parameters should cover regular usage and don't confuse users.

It is some kind of logging utilities. Even in functional world nobody brings logger instance parameter into each function.